src/webextension/list/filter.js

```diff
@@ -7,29 +7,56 @@ export function parseFilterString(filter) {
   if (!filter) {
     return [];
   }
-  return filter.split(/\s+/).map((i) => i.toLocaleLowerCase());
+  return filter.split(/\s+/);
+}
+
+function tryMakeURL(string, defaultProtocol = null) {
+  let url = null;
+  try {
+    url = new URL(string);
+  } catch (e) {
+    if (defaultProtocol) {
+      try {
+        url = new URL(defaultProtocol + "//" + string);
+      } catch (e) {}
+    }
+  }
+  return {url, string};
 }
 
 function match(filterElement, value) {
-  return value.includes(filterElement);
+  return value.toLocaleLowerCase().includes(filterElement.toLocaleLowerCase());
+}
+
+function matchURL(filterElement, value) {
+  // If the user is in the middle of typing a URL or entered something not
+  // URL-like in the origin field, just to a string comparison. This is safe for
+  // strictly matching auto-filled URLs since they're always full hosts with
+  // protocol (and possibly port).
+  if (!filterElement.url || !value.url) {
+    return match(filterElement.string, value.string);
+  }
+  return (filterElement.url.protocol === value.url.protocol &&
+          filterElement.url.host === value.url.host);
```

Comment:
looking at this more, I'm less and less comfortable with the missing port.

Reply:
`URL.host` includes the port if it's not the default port for that protocol. So for `https://www.google.com`, `URL.host` is `www.google.com`, but for `https://www.google.com:8000`, `URL.host` is `www.google.com:8000`. I plan to make this a lot more explicit when I build some common functions to use here (i.e. I'll write documentation).

Reply:
I apologize for that; I keep having them swapped in my head.

Reply:
No worries; this is why we need documentation (and instinctively, `hostname` is longer than `host`, so the value seems like it should be longer too!).